src/rhsmlib/services/unregister.py

```diff
@@ -65,3 +69,12 @@ def unregister(self):
                 managerlib.clean_all_data(backup=False)
             else:
                 raise ge
+        finally:
+            try:
+                if os.path.exists(INSIGHTS_REGISTER_UNIT_PATH) and os.path.islink(INSIGHTS_REGISTER_UNIT_PATH) and \
```

Comment:
Should we always unmask on unregister? Can an insights registration be considered valid without the associated subscription-manager identity?

What about insights registrations performed using username and password? Should we still unmask the path unit in the case of an insights registration which is independent of subscription-manager?
I think if there is a insights registration that is still valid we should not unmask.

CC: @subpop

Reply:
> Should we always unmask on unregister? Can an insights registration be considered valid without the associated subscription-manager identity?

No. While the host record may exist in HBI for a period of time after registration, it will not be possible for a host to submit new archives to the platform services. Eventually, the stale host record will get deleted from the database. A host in this state is not considered active, since it is unable to perform successful API calls back to platform services.

> What about insights registrations performed using username and password? Should we still unmask the path unit in the case of an insights registration which is independent of subscription-manager?
> I think if there is a insights registration that is still valid we should not unmask.

I think it is safe to always unmask the `insights-register.path` unit when `subscription-manager` unregisters a host. If a host is registered with Insights at the time `/etc/pki/consumer/cert.pem` disappears, the `insights-unregister.path` will trigger and will clean up the local Insights registration files (this leaves the host in the stale state described above). Without a certificate, automatic communication with the platform services is not possible.

Reply:
A slight addendum to my earlier statement. It is possible, however unlikely, that a host may be registered with RHSM, and registered with Insights using basic authentication. In order for this to be possible, the system administration must have written their username and password into `/etc/insights-client/insights-client.conf` in clear text. Assuming this obvious security hole is permitted by their organization's policy, if a user runs `subscription-manager unregister` on that host, then the resulting execution of `insights-client --unregister` that will occur as a result of the `insights-unregister.path` unit triggering will successfully unregister the host from Insights in addition to cleaning up the local registration data.